### PR TITLE
Uf 5930 rename and extend service

### PR DESCRIPTION
**Pull request by josc04**
- Added new DB attributes for origin and party type
- Extended API with possibility to provide origin of asset on create
- Extended API to return origin of asset on read
- Added integration to party service
- Extended with logic to calculate party type (private or enterprise)
for party connected to asset

### CODE REVIEW

File: src/main/java/se/sundsvall/partyassets/service/AssetService.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		return repository.save(toEntity(request, calculatePartyType(request.getPartyId()))).getId();
+	}
+
+	private PartyType calculatePartyType(String partyId) {
+		return partyClient.getLegalId(generated.se.sundsvall.party.PartyType.PRIVATE, partyId)
+			.isPresent() ? PRIVATE : ENTERPRISE;
+	}
+
+	public void deleteAsset(final String id) {
+		repository.deleteById(id);
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/partyassets/service/AssetService.java:61]: 404 if entity doesn't exist?

File: src/main/java/se/sundsvall/partyassets/service/AssetService.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				.withTitle("Asset already exists")
+				.withDetail("Asset with assetId %s already exists".formatted(request.getAssetId()))
+				.build();
+		}
+		return repository.save(toEntity(request, calculatePartyType(request.getPartyId()))).getId();
+	}
+
+	private PartyType calculatePartyType(String partyId) {
+		return partyClient.getLegalId(generated.se.sundsvall.party.PartyType.PRIVATE, partyId)
+			.isPresent() ? PRIVATE : ENTERPRISE;
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/partyassets/service/AssetService.java:57]: If "dummy" is sent to this method, "ENTERPRISE" will be returned. Is this an appropriate behaviour?

**Comment by foobar-1337** [comment]
Fix existing codesmell?

File: src/main/java/se/sundsvall/partyassets/integration/party/PartyClient.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import generated.se.sundsvall.party.PartyType;
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+import se.sundsvall.partyassets.integration.party.configuration.PartyConfiguration;
+
+@CircuitBreaker(name = CLIENT_ID)
```
Comment (foobar-1337) [inline src/main/java/se/sundsvall/partyassets/integration/party/PartyClient.java:17]: @CircuitBreaker not necessary here

File: src/main/java/se/sundsvall/partyassets/service/AssetService.java (hunk 1 above)
Comment (josc04) [reply src/main/java/se/sundsvall/partyassets/service/AssetService.java:61]: Yessir

**Comment by josc04** [comment]
> Fix existing codesmell?

Ofcourse, fixed!

File: src/main/java/se/sundsvall/partyassets/service/AssetService.java (hunk 2 above)
Comment (josc04) [reply src/main/java/se/sundsvall/partyassets/service/AssetService.java:57]: This attribute is not requested by the stakeholders and is not visible in the API, it's more of a "good to have" thing. Throwing exception for non existing partyIds (which will probably never happen) might not be aligned with the requirements from the stakeholders.

Another option is to add a third partyType, UNKNOWN, if id isn't recognized as private or enterprise type.

I'll check with Jakob on how to handle if/when type for partyId is not recognized. TBC.

Comment (josc04) [reply src/main/java/se/sundsvall/partyassets/service/AssetService.java]: Solution is to go with your recommendation and throw fault if party type can not be calculated, this is now changed.

**Comment by josc04** [comment]
Acted on all PR-comments

**Pull request merged**